thrust/system/cuda/detail/core/util.h

```diff
@@ -652,7 +652,10 @@ namespace core {
   }
 
 #define CUDA_CUB_RET_IF_FAIL(e) \
-  if (cub::Debug((e), __FILE__, __LINE__)) return e;
+  do {                          \
+    auto const error = (e);     \
+    if (cub::Debug(error, __FILE__, __LINE__)) return error; \
+  } while(0);
 
```

Comment:
Nitpick: The `do..while` isn't necessary here, this can just be 

```
{ \
    auto const error = (e);     \
    if (cub::Debug(error, __FILE__, __LINE__)) \
      return error; \
}
```

Reply:
@allisonvacanti  There are cases where using a scope block causes issues, please see https://stackoverflow.com/a/1067238

Reply:
For that to work, you can't have the semicolon after the `while (0)`.  With that semicolon there, it has the same problem as if it were a block scope.

Reply:
To update this thread with a slack convo, we don't want that behavior here since the old version didn't require a trailing semicolon. Using the `do/while(0)` pattern (without the semicolon) risks breaking other code.